src/components/Homepage/index.js

```diff
@@ -1,15 +1,27 @@
-import {useEffect} from "react";
+import {useEffect, useState} from "react";
+import Button from '../Button'
 
-const Homepage = ({restaurantsData, updateRestaurantsData}) => {
-    function fetchRestaurants() {
+const Homepage = () => {
+
+    const [restaurantsData, setRestaurantsData] = useState([])
+
+    useEffect(() => {
         fetch("http://localhost:8080/restaurants")
             .then(response => response.json())
             .then(data => {
-                updateRestaurantsData(data)
+                setRestaurantsData(data)
             })
-    }
+    }, [])
+
+    let restaurants = restaurantsData.map((restaurant) => {
+        return <Button key={restaurant.id} name={restaurant.name} />
+    })
 
-    useEffect(() => {fetchRestaurants()}, [])
+    return (
+        <>
```

Comment:
The main tag currently coming from App should be here - then you can ditch the fragment tags